src/compiler/emitter.ts

```diff
@@ -3052,6 +3054,17 @@ module ts {
                     write(": ");
                     // Even though this is stored as identifier treat it as an expression
                     // Short-hand, { x }, is equivalent of normal form { x: x }
+
+                    if (languageVersion < ScriptTarget.ES6) {
+                    // short-hand property can be initialized with renamed let binding
+                    // in this case use proper name
+                        var generatedName = getGeneratedNameForIdentifier(node.name, /*isValueOfShorthandPropertyAssignment*/ true);

```

Comment:
Correct me if i'm wrong, but if we have:  ```{ id }```    then it will always be rewritten as ```{ id: ... }```    So we always can just write out the 'id' as is initially.  The question is only what we need to write after the colon.  And, in that case, we don't need to pass this flag along stating if we're the value side of the shorthand property assignment.